packages/gasket-intl/README.md

```diff
@@ -10,291 +20,175 @@ npm i @gasket/intl
-
-The parts can be described in the following shapes:
-- string: `<module>.<namespace>`
-- string[]: array of strings in the form `<module>.<namespace>`
-- object: `{module, namespace}`
-- object[]: array of objects with the form `{module, namespace}`
-
-## Component Props
+### withIntlProvider
 
-- `identifier`: (Identifier) What locale file(s) to load.
-- `loading`: (string|node) Optional content to render while loading. Defaults to
-  null. When using the HOC at the page-level, `getInitialProps` will be invoked
-  and defer rendering until locale content is loaded.
+Use this to wrap an app to set up the Provider for [react-intl] and context for
+the locale components below.
 
-## HOC Arguments
+**Signature**
 
-- `identifier`: (Identifier) What locale file(s) to load.
-- `otherProps`: (Object) Besides identifier, other props supported by the
-  component.
+- `withIntlProvider(options)`
 
-## Selectors
+**Props**
 
-### selectMessage
+- `[options]` - (object) Optional configuration - currently not used
 
-`selectMessage(state, id, [defaultMessage])`
+```jsx
+import { withIntlProvider } from '@gasket/intl';
 
-For use outside of `react-intl` provider context.
+const App = props => <div>{props.children}</div>
 
-- `state`: (object) redux state
-- `id`: (string) message id or key
-- `defaultMessage`: (string) Optional message if `id` is not found. Otherwise
-  the name of the id will be returned.
+export default withIntlProvider()(App);
+```
 
-### selectAllMessages
+### withLocaleRequired
 
-`selectAllMessages(state)`
+Higher-order component to wrap pages or components in an app. This checks to see
+if a locale file has been loaded, and fetches it if not. Once loaded, the
+wrapped component will be rendered.
 
-Used by IntlProvider internally, but exposed for some edge use cases.
+**Signature**
 
-## Usage
+- `withLocaleRequired(localesPath, options)`
 
-### LocaleRequired
+**Props**
 
-This component loads localization messages for the associated package as
-described by the identifier.
+- `localesPath` - (string) Path to endpoint with JSON files. See more about
+  [locales path] in the plugin docs.
+- `[options]` - (object) Optional configuration
+  - `loading` - (string|node) Content to render while loading, otherwise null.
 
-#### Example : When keys are defined in the `locales` root.
+#### Example
 
-In the below example, `app_more_info` and `app_details` are two keys defined in
-the locale file for our app. If `identifier` prop is not set, it will default to
-the locale file of the app.
+```jsx
+import { withLocaleRequired } from '@gasket/intl';
+import { FormattedMessage } from 'react-intl';
 
-```js
-import { LocaleRequired } from '@gasket/intl';
+const Component = props => <h1><FormattedMessage id='welcome'/></h1>
 
-<LocaleRequired>
-  <div>
-    <div><FormattedMessage id='app_more_info' /></div>
-    <div><FormattedMessage id='app_details' /></div>
-  </div>
-</LocaleRequired>
+export default withLocaleRequired('/locales')(Component);
```

Comment:
is the localesPath parameter mandatory?

Reply:
No. I add docs and an example describing using a default localesPath.